dashboard/db/migrate/20230713141627_add_indexes_to_rubric_tables.rb

```diff
@@ -0,0 +1,8 @@
+class AddIndexesToRubricTables < ActiveRecord::Migration[6.1]
+  def change
+    add_index :rubrics, :lesson_id, unique: true
+    add_index :rubrics, :level_id
```

Comment:
Do we want the index on the lesson and then the lesson + level? Would we ever want ALL rubrics for a level or just the ONE that matches the context?

Reply:
I think you're right that we want an index on lesson id and level id.

The thing I'm trying to work through now is that we never want more than one rubric for a lesson. I think there are two ways to handle this:
1. The unique index on lesson id that I have here
2. Requiring both a lesson id and a level id be non-null and relying on the unique index on both lesson id and level id

It was a miss when I created these tables to not add not null constraints on these columns. That said, I'm struggling with whether we would ever want level_id to be null. The scenario I see is that a curriculum writer creates a lesson and already knows what the rubric should be so they add that then go back and add the level. But that feels like an inefficient process and one that could lead to forgetting to add the level back in, so I'm tempted to go with option 2. Thoughts?

Reply:
(I pushed a commit for option 2 just now. Feel free to push back -- I'm just about to get my local environment into a weird state and wanted to push what I had)

Reply:
I agree with the idea that there must be a level.

The scenario I see is one where the curriculum writer actually does create the rubric on the level that would be assessed. The lesson is tagged only so that it may be visible in places where the lesson is rendered or on preceding levels. (The interesting bit is whether or not rubrics are shown on levels AFTER the assessment. 😬🤔)

So, they would have the option of adding a rubric to an added level of the lesson. They would have to add the level and then add the rubric.

Reply:
I thought that lesson would be required no matter what so I was thinking of this in the opposite way -- it's possible to have a rubric on a lesson without having created the level yet.

Reply:
Either way, I think we're in agreement that requiring both is the right direction, so I think we're splitting hairs a bit. I'll keep these as is -- we can always make these less restrictive in the future!